### ISSUE TITLE
Suppport for gamemode on Linux

### ISSUE THREAD

**Issue by stephanlachnit** [tomatenquark/code#37]
It would be cool if Tomatenquark could add support for [gamemode](https://github.com/FeralInteractive/gamemode). Basically what it does it changing the CPU gov for the game (and some other performance tricks), and resetting it afterwards, so it should increase performance a bit for some users.
It can be integrated in three ways:
1. Include a header in the main function
2. Statically link libgamemodeauto (basically the same as 1.)
3. Dynamically link libgamemodeauto

Since Make isn't exactly very flexible when it comes to build options (it's possible but ugly), I would wait for #22 to be resolved.
In Meson, it would be quite simple to give all these three options. At least having both 1. and 3. is nice because that means that one can simply add it for a Steam build without worrying if gamemode is installed, but also Distro shipping Tomatenquark (maybe I'll go for it in Debian if I find the time) can simply link against the system installation of libgamemodeauto.

**Comment by Calinou** [comment]
Considering Tomatenquark's low system requirements, would enabling GameMode help meaningfully? I guess it doesn't hurt to do it, but we should weigh in the maintenance cost of integrating GameMode.

**Comment by stephanlachnit** [comment]
> Considering Tomatenquark's low system requirements, would enabling GameMode help meaningfully? I guess it doesn't hurt to do it, but we should weigh in the maintenance cost of integrating GameMode.

I don't think it would make a huge difference, but the client integration is really really simple and failsafe (tries to load gamemode if installed on the system, else it does nothing). [It's possible to do it in two lines](https://github.com/FeralInteractive/gamemode/blob/master/lib/README.md):
```C
#define GAMEMODE_AUTO
#include "gamemode_client.h"
```

With a build option that would extend to

```C
#ifdef USE_GAMEMODE
#define GAMEMODE_AUTO
#include "gamemode_client.h"
#endif
```

**Comment by Fohlen** [comment]
I'd say, given #22 would be merged, I don't see why we should not go for it.

**Comment by Fohlen** [comment]
@stephanlachnit with #40 becoming merged would you like to do the following:

- create a `vcpkg` port for `gamemode`
- add a PR for linux using this

All the best

**Comment by stephanlachnit** [comment]
> create a vcpkg port for gamemode

I don't think that is necessary, but maybe I'll look into adding a build option. After all, it's literally just one header and four lines.

**Comment by Fohlen** [comment]
@stephanlachnit I do not like including "just one header". It becomes messy quickly. If gamemode uses CMake we can also use the `FetchContent` module like so: https://github.com/tomatenquark/code/blob/master/CMakeLists.txt#L71.
Anyways. Could you create a PR for this?

**Comment by stephanlachnit** [comment]
I'm sorry right now I don't really have time for this
